lib/services_filter.rb

```diff
@@ -0,0 +1,35 @@
+require 'exceptions'
+
+class ServicesFilter
+  class << self
+    delegate :call, to: :new
+  end
+
+  def initialize(model_class = Location)
+    @model_class = model_class
+  end
+
+  def call(category_ids)
+    if category_ids.blank? || category_ids.empty?
+      @model_class.all
+    else
+      @model_class.where(id: filter_by_categories(category_ids))
+    end
+  end
+
+  private
+
+  def filter_by_categories(category_ids)
+    locations = @model_class.select do |location|
```

Comment:
This works, but would be a good candidate for refactoring since it hammers the database pretty hard.